presto-main/src/main/java/com/facebook/presto/cost/CachingStatsProvider.java

```diff
@@ -0,0 +1,93 @@
+/*
```

Comment:
Can you add more info to the commit message about the api change?

Reply:
sure, i can. What you think is most important? Also, I believe there are no users of the current API (outside of CBO that is), so it's hard for me to determine what's important.